### PR TITLE
Adds execution context flow options

### LINKED ISSUE #25: ExecutionContext is not restored after the server is disposed
In a large test-suite where we rely on `ExecutionContext` to be able to flow we have noticed that running `Nowin` breaks our tests. We have identified the source of the problem in [SaeaLayerCallback.cs](https://github.com/Bobris/Nowin/blob/master/Nowin/SaeaLayerCallback.cs#L206).

Please consider wrapping the suppression in a using statement at the correct level so that the async methods don't flow context but the main thread can retain the context flowing for the rest of the application.

### PR DESCRIPTION
**Pull request by serialseb**
This should help fix #25. Added a bunch of options one can chose from. I don't see much change in processing response times using the benchmarks, i get between 3000 to 3100/s on all options (and the same on the original code). Let me know if the memory allocs are equivalent or if you want to see something else in there.

### CODE REVIEW

**Comment by Bobris** [comment]
You need to disable Windows Defender before bench-marking. My results on not much beefy notebook:

No keepalive | keepalive | settings
-------- | --------- | -----------------------
10300 | 60615 | Previous version
10288 | 60132 | SuppressAlways
10210 | 61180 | SuppressOnAsync
 9112 | 59497 | Flow

SuppressAlways and SuppressOnAsync are basically equal. And there is not measurable slow down from previous code.

Building release as you reading this.

**Comment by Bobris** [comment]
0.14.0 version on nuget.

File: Nowin/SaeaLayerCallback.cs [hunk 1]
```diff
@@ -203,10 +206,9 @@ public void StartAccept(byte[] buffer, int offset, int length)
             }
         }
 
-        void StopExecutionContextFlow()
-        {
-            if (!ExecutionContext.IsFlowSuppressed())
-                ExecutionContext.SuppressFlow();
+        IDisposable StopExecutionContextFlow()
+        {;
```
Comment (davidfowl) [inline Nowin/SaeaLayerCallback.cs]: Extra semicolon here

Comment (Bobris) [inline Nowin/SaeaLayerCallback.cs]: Fixed. Thanks.

**Pull request merged**